src/Microsoft.ML.LightGbm/LightGbmMulticlassTrainer.cs

```diff
@@ -255,6 +266,13 @@ private protected override void ConvertNaNLabels(IChannel ch, RoleMappedData dat
                     _tlcNumClass = (int)maxLabel + 1;
                 }
             }
+
+            // If there are NaN labels, they are converted to be equal to _tlcNumClass (i.e. _numClass - 1).
```

Comment:
I am not sure if this comment I added is clear enough, or if I should have added it somewhere else, or if I shouldn't add it anyway.

Problem is that @yaeldekel and I think that having both _numClass and _tlcNumClass variables  is somewhat confusing and it's not clear why they exist... and simply adding a comment somewhere (explaining their difference) might be useful.

Reply:
I would put this comment where the fields are defined instead.
Also, maybe we should rename the fields?
`_tlcNumClass` -> `_numberOfClasses`
`_numClass` => `_numberOfClassesIncludingNaN` (or `_lightGbmNumberOfClasses` - this name in addition to your comment explaining that LightGBM considers NaN to be an additional class might be clear enough).

---
In reply to: [362128258](https://github.com/dotnet/machinelearning/pull/4608#discussion_r362128258) [](ancestors = 362128258)